mailserver/mailserver.go

```diff
@@ -288,24 +290,56 @@ func (s *WMailServer) DeliverMail(peer *whisper.Peer, request *whisper.Envelope)
 	}
 
 	log.Debug("Processing request",
-		"lower", lower, "upper", upper,
+		"lower", lower,
+		"upper", upper,
 		"bloom", bloom,
 		"limit", limit,
 		"cursor", cursor,
-		"batch", batch)
+		"batch", batch,
+	)
 
 	if batch {
 		requestsBatchedCounter.Inc(1)
 	}
 
-	_, lastEnvelopeHash, nextPageCursor, err := s.processRequest(
-		peer,
-		lower, upper,
+	iter := s.createIterator(lower, upper, cursor)
+	defer iter.Release()
+
+	bundles := make(chan []*whisper.Envelope, 5)
+	errCh := make(chan error)
+
+	go func() {
+		for bundle := range bundles {
+			if err := s.sendEnvelopes(peer, bundle, batch); err != nil {
+				errCh <- err
+				break
+			}
+		}
+		close(errCh)
+	}()
+
+	start := time.Now()
+	nextPageCursor, lastEnvelopeHash := s.processRequestInBundles(
```

Comment:
Some stress tests for this fragment would be nice. We use buffered channel but if there are more than 5 bundles, sending them through the network will become a bottleneck. It is still better than the previous implementation which was sending bundles sequentially.